# ADS 505 Final Project

## Best Home Regions

### Claire Bentzen, Jiaqi He, S M Sultan Mahmud Rahat

## 1. Data Preprocessing

In [2]:
import pandas as pd

In [42]:
# read in data
homes = pd.read_csv('final_dataset.csv')
homes.head()

MortgageRateConventionalFixed_2011-06  \
0                               4.374217   
1                               4.374217   
2                               4.374217   
3                               4.374217   
4                               4.374217   

   MortgageRateConventionalFixed_2011-07  \
0                               4.460115   
1                               4.460115   
2                               4.460115   
3                               4.460115   
4                               4.460115   

   MortgageRateConventionalFixed_2011-08  \
0                               4.127827   
1                               4.127827   
2                               4.127827   
3                               4.127827   
4                               4.127827   

   MortgageRateConventionalFixed_2011-09  \
0                               3.963005   
1                               3.963005   
2                               3.963005   
3                               3.963005   
4                               3.963005   

   MortgageRateConventionalFixed_2011-10  \
0                               4.035497   
1                               4.035497   
2                               4.035497   
3                               4.035497   
4                               4.035497   

   MortgageRateConventionalFixed_2011-11  \
0                               3.941695   
1                               3.941695   
2                               3.941695   
3                               3.941695   
4                               3.941695   

   MortgageRateConventionalFixed_2011-12  \
0                                3.85004   
1                                3.85004   
2                                3.85004   
3                                3.85004   
4                                3.85004   

   MortgageRateConventionalFixed_2012-01  \
0                               3.815944   
1                               3.815944   
2                               3.815944   
3                               3.815944   
4                               3.815944   

   MortgageRateConventionalFixed_2012-02  \
0                               3.808312   
1                               3.808312   
2                               3.808312   
3                               3.808312   
4                               3.808312   

   MortgageRateConventionalFixed_2012-03  ...  PriceToRentRatio_2017-07  \
0                               3.931269  ...                     10.59   
1                               3.931269  ...                     12.77   
2                               3.931269  ...                     11.26   
3                               3.931269  ...                     12.96   
4                               3.931269  ...                     11.97   

   PriceToRentRatio_2017-08  PriceToRentRatio_2017-09  RegionID  RegionName  \
0                     10.64                     10.72     58196        1001   
1                     12.94                     13.15     58197        1002   
2                     11.36                     11.50     58200        1005   
3                     12.94                     12.77     58201        1007   
4                     11.98                     11.97     58202        1008   

                  City  State        Metro  CountyName  SizeRank  
0               Agawam     MA  Springfield     Hampden      5770  
1              Amherst     MA  Springfield   Hampshire      4112  
2        Town of Barre     MA    Worcester   Worcester     11428  
3  Town of Belchertown     MA  Springfield   Hampshire      6855  
4            Blandford     MA  Springfield     Hampden     15088  

[5 rows x 167 columns]

In [43]:
# identify number of rows and columns
homes.shape

(14233, 167)

#### Check for Duplicate Records

In [44]:
# check for duplicates
duplicate_homes = homes[homes.duplicated()]
duplicate_homes.shape

(0, 167)

There are no duplicate records in the dataset.

#### Check for Missing Values

In [45]:
# check for overall percentage of missing values
missing_values = homes.isnull().sum().sum()
percent_missing = (missing_values / homes.size) * 100
print("Total percentage of missing data: {:.2f}%".format(percent_missing))

Total percentage of missing data: 1.64%


In [60]:
pd.set_option('display.max_rows', None)

# check which columns missing values reside in
missing_homes = homes.isnull().sum()
columns_missing_values = missing_homes[missing_homes > 0]

# turn missing values series into dataframe
columns_missing_values_df = columns_missing_values.to_frame()
columns_missing_values_df.columns = ['Count']

In [61]:
# missing values as percentages
columns_missing_values_percents = (columns_missing_values / homes.size) * 100

# turn missing values percentage series into dataframe
columns_missing_values_percents_df = columns_missing_values_percents.to_frame()
columns_missing_values_percents_df.columns = ['%']

In [63]:
# merge dataframes to very counts against percentages
missing = columns_missing_values_df.join(columns_missing_values_percents_df, how='inner')
missing

Count         %
PriceToRentRatio_2010-10    994  0.041819
PriceToRentRatio_2010-11   1184  0.049813
PriceToRentRatio_2010-12    906  0.038117
PriceToRentRatio_2011-01    691  0.029071
PriceToRentRatio_2011-02    651  0.027388
PriceToRentRatio_2011-03    724  0.030460
PriceToRentRatio_2011-04    635  0.026715
PriceToRentRatio_2011-05    644  0.027094
PriceToRentRatio_2011-06    693  0.029155
PriceToRentRatio_2011-07    632  0.026589
PriceToRentRatio_2011-08    608  0.025579
PriceToRentRatio_2011-09    609  0.025621
PriceToRentRatio_2011-10    607  0.025537
PriceToRentRatio_2011-11    645  0.027136
PriceToRentRatio_2011-12    683  0.028735
PriceToRentRatio_2012-01    633  0.026631
PriceToRentRatio_2012-02    591  0.024864
PriceToRentRatio_2012-03    617  0.025958
PriceToRentRatio_2012-04    617  0.025958
PriceToRentRatio_2012-05    632  0.026589
PriceToRentRatio_2012-06    679  0.028566
PriceToRentRatio_2012-07    664  0.027935
PriceToRentRatio_2012-08    701  0.029492
PriceToRentRatio_2012-09    636  0.026757
PriceToRentRatio_2012-10    670  0.028188
PriceToRentRatio_2012-11    602  0.025327
PriceToRentRatio_2012-12    619  0.026042
PriceToRentRatio_2013-01    628  0.026421
PriceToRentRatio_2013-02    608  0.025579
PriceToRentRatio_2013-03    641  0.026968
PriceToRentRatio_2013-04    579  0.024359
PriceToRentRatio_2013-05    529  0.022256
PriceToRentRatio_2013-06    504  0.021204
PriceToRentRatio_2013-07    505  0.021246
PriceToRentRatio_2013-08    446  0.018764
PriceToRentRatio_2013-09    446  0.018764
PriceToRentRatio_2013-10    436  0.018343
PriceToRentRatio_2013-11    435  0.018301
PriceToRentRatio_2013-12    472  0.019858
PriceToRentRatio_2014-01    471  0.019816
PriceToRentRatio_2014-02    457  0.019227
PriceToRentRatio_2014-03    454  0.019100
PriceToRentRatio_2014-04    437  0.018385
PriceToRentRatio_2014-05    467  0.019647
PriceToRentRatio_2014-06    404  0.016997
PriceToRentRatio_2014-07    406  0.017081
PriceToRentRatio_2014-08    404  0.016997
PriceToRentRatio_2014-09    404  0.016997
PriceToRentRatio_2014-10    404  0.016997
PriceToRentRatio_2014-11    404  0.016997
PriceToRentRatio_2014-12    406  0.017081
PriceToRentRatio_2015-01    414  0.017418
PriceToRentRatio_2015-02    401  0.016871
PriceToRentRatio_2015-03    401  0.016871
PriceToRentRatio_2015-04    413  0.017375
PriceToRentRatio_2015-05    401  0.016871
PriceToRentRatio_2015-06    402  0.016913
PriceToRentRatio_2015-07    435  0.018301
PriceToRentRatio_2015-08    397  0.016702
PriceToRentRatio_2015-09    397  0.016702
PriceToRentRatio_2015-10    396  0.016660
PriceToRentRatio_2015-11    391  0.016450
PriceToRentRatio_2015-12    383  0.016113
PriceToRentRatio_2016-01    383  0.016113
PriceToRentRatio_2016-02    413  0.017375
PriceToRentRatio_2016-03    396  0.016660
PriceToRentRatio_2016-04    389  0.016366
PriceToRentRatio_2016-05    387  0.016282
PriceToRentRatio_2016-06    272  0.011443
PriceToRentRatio_2016-07    218  0.009172
PriceToRentRatio_2016-08    255  0.010728
PriceToRentRatio_2016-09    217  0.009129
PriceToRentRatio_2016-10    217  0.009129
PriceToRentRatio_2016-11    217  0.009129
PriceToRentRatio_2016-12    260  0.010939
PriceToRentRatio_2017-01    216  0.009087
PriceToRentRatio_2017-02    214  0.009003
PriceToRentRatio_2017-03    213  0.008961
PriceToRentRatio_2017-04     38  0.001599

In [64]:
# drop missing values in the dataset
homes = homes.dropna()
homes.shape

(12786, 167)

Although missing values only made up 1.64% of the overall dataset, we have dropped rows that contained missing values which brought the dataset to have 12,786 remaining records.

## 2. Exploratory Data Analysis